dbinspector/src/main/kotlin/im/dino/dbinspector/ui/shared/delegates/ViewBindingDelegate.kt

```diff
@@ -0,0 +1,62 @@
+package im.dino.dbinspector.ui.shared.delegates
+
+import android.view.LayoutInflater
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.viewbinding.ViewBinding
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+internal class ViewBindingDelegate<T : ViewBinding>(
+    val fragment: Fragment,
+    val viewBindingFactory: (View) -> T
+) : ReadOnlyProperty<Fragment, T> {
+
+    private var binding: T? = null
+
+    init {
+        fragment.lifecycle.addObserver(
+            object : DefaultLifecycleObserver {
+                override fun onCreate(owner: LifecycleOwner) {
+                    fragment.viewLifecycleOwnerLiveData.observe(fragment) { viewLifecycleOwner ->
+                        viewLifecycleOwner.lifecycle.addObserver(
+                            object : DefaultLifecycleObserver {
+                                override fun onDestroy(owner: LifecycleOwner) {
+                                    binding = null
+                                }
+                            }
+                        )
+                    }
+                }
+            }
+        )
```

Comment:
This is a bit of a mess to look at 😄 maybe it makes sense to extract some of those objects into variables so it's easier on the eyes

Reply:
@JonatanPlesko To me, this is perfectly fine since this delegate should be a library itself. Now it's just copy pasted around.

Reply:
The fact it's being copy pasted around just makes it worse 😄 but it's ok, I'm not too bothered

Reply:
@JonatanPlesko Find someone to make a lib out of that class :) 
I am dead serious.